### PR TITLE
Add `updateImageSource`.

### PR DESCRIPTION
**Pull request by CaviarChen**
This PR basically backports an existing function from [https://github.com/flutter-mapbox-gl/maps](https://github.com/flutter-mapbox-gl/maps).
Unlike that one, this does not suffer from https://github.com/mapbox/mapbox-maps-android/issues/302 and it works nicely on both iOS and Android. The web version is still unimplemented, similar to existing image source related functions.
(Sorry a thing I'm building only requires `updateImageSource` on both iOS and Android.)

### CODE REVIEW

**Comment by m0nac0** [comment]
@CaviarChen Thanks for the contribution! I haven't had time to test this, yet, but the code looks good 👍

**Comment by CaviarChen** [comment]
merged from the upstream and tested, everything works fine.

**Comment by CaviarChen** [comment]
A gentle ping on this.

File: lib/src/controller.dart [hunk 1]
```diff
@@ -1086,6 +1086,13 @@ class MaplibreMapController extends ChangeNotifier {
     return _mapboxGlPlatform.addImageSource(imageSourceId, bytes, coordinates);
   }
 
+  /// Update the image and/or coordinates of an image source.
```
Comment (m0nac0) [inline lib/src/controller.dart:1090]:
```suggestion
  /// Update the image and/or coordinates of an image source.
  /// Not implemented on web.
```

Comment (m0nac0) [reply lib/src/controller.dart:1090]: Could you add this doc comment about web support (and maybe also to the addImageSource method)?

**Comment by m0nac0** [comment]
Sorry, I haven't had time to actually test this. I left a comment about the documentation. 
And could you also add an example for the new method? That makes testing easier. Could e.g. be just one more button in the existing Place Source page in the example app.

**Comment by CaviarChen** [comment]
Changed as requested. However, I do spot an issue and I don't think I know this lib enough to know where went wrong. It seems that the `Change image` button works fine on iOS but on Android, I need to touch (move) the map to trigger a refresh  to make the image change.

**Comment by CaviarChen** [comment]
Played with the example a bit and I think the bug is not introduced by my PR.
Reproduce:
open the example app and go to the place source page,
click add source
then click show layer
you can see the image show up as expected
then click hide layer
the image does not go away until you do something like touch the map (which trigger a redraw?)

**Comment by CaviarChen** [comment]
ping

**Comment by m0nac0** [comment]
Thank you for your patience!
LGTM! I just noticed that the images appear flipped in the iOS simulator, but that doesn't seem to be related to your work.

Just one more question: is the new asset file your own picture or licensed in a way that's compatible with the repo license?

**Comment by CaviarChen** [comment]
Ah that's a good question, I might just found it from google. I have no experence in this, do you have any recommanded way of finding pic for this?

**Comment by m0nac0** [comment]
I guess you could search on wikimedia commons and use e.g. a CC0 licensed image

e.g. https://commons.wikimedia.org/w/index.php?search=sydney&title=Special:MediaSearch&go=Go&type=image&haslicense=unrestricted

**Comment by CaviarChen** [comment]
Replaced the image, the new one is found here: https://commons.wikimedia.org/wiki/File:Sydney_Opera_House_(2017).jpg

**Comment by m0nac0** [comment]
Thank you again for your contribution and patience!

**Review by m0nac0 — APPROVED** [review]

**Pull request merged**